#!/usr/bin/env python
PACKAGE = "object_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("R_H_min",    	int_t,    0, "Hue min", 105,  0, 255)
gen.add("R_H_max",    	int_t,    0, "Hue max", 130,  0, 255)
gen.add("R_S_min",    	int_t,    0, "Saturation min", 100,  0, 255)
gen.add("R_S_max",    	int_t,    0, "Saturation max", 255,  0, 255)
gen.add("R_V_min",    	int_t,    0, "Value min", 1,  0, 255)
gen.add("R_V_max",    	int_t,    0, "Value max", 255,  0, 255)
gen.add("R_Display", bool_t,   0, "Display mask",  False)

gen.add("B_H_min",    	int_t,    0, "Hue min", 5,  0, 255)
gen.add("B_H_max",    	int_t,    0, "Hue max", 25,  0, 255)
gen.add("B_S_min",    	int_t,    0, "Saturation min", 1,  0, 255)
gen.add("B_S_max",    	int_t,    0, "Saturation max", 255,  0, 255)
gen.add("B_V_min",    	int_t,    0, "Value min", 1,  0, 255)
gen.add("B_V_max",    	int_t,    0, "Value max", 255,  0, 255)
gen.add("B_Display", bool_t,   0, "Display mask",  False)

gen.add("G_H_min",    	int_t,    0, "Hue min", 50,  0, 255)
gen.add("G_H_max",    	int_t,    0, "Hue max", 85,  0, 255)
gen.add("G_S_min",    	int_t,    0, "Saturation min", 1,  0, 255)
gen.add("G_S_max",    	int_t,    0, "Saturation max", 255,  0, 255)
gen.add("G_V_min",    	int_t,    0, "Value min", 1,  0, 255)
gen.add("G_V_max",    	int_t,    0, "Value max", 255,  0, 255)
gen.add("G_Display", bool_t,   0, "Display mask",  False)

gen.add("Y_H_min",    	int_t,    0, "Hue min", 85,  0, 255)
gen.add("Y_H_max",    	int_t,    0, "Hue max", 105,  0, 255)
gen.add("Y_S_min",    	int_t,    0, "Saturation min", 100,  0, 255)
gen.add("Y_S_max",    	int_t,    0, "Saturation max", 250,  0, 255)
gen.add("Y_V_min",    	int_t,    0, "Value min", 1,  0, 255)
gen.add("Y_V_max",    	int_t,    0, "Value max", 255,  0, 255)
gen.add("Y_Display", bool_t,   0, "Display mask",  False)

gen.add("P_H_min",    	int_t,    0, "Hue min", 140,  0, 255)
gen.add("P_H_max",    	int_t,    0, "Hue max", 170,  0, 255)
gen.add("P_S_min",    	int_t,    0, "Saturation min", 1,  0, 255)
gen.add("P_S_max",    	int_t,    0, "Saturation max", 255,  0, 255)
gen.add("P_V_min",    	int_t,    0, "Value min", 1,  0, 255)
gen.add("P_V_max",    	int_t,    0, "Value max", 255,  0, 255)
gen.add("P_Display", bool_t,   0, "Display mask",  False)

gen.add("Remove_floor", bool_t,   0, "Remove floor",  True)

exit(gen.generate(PACKAGE, "object_detection", "ColorTuning"))
